akka-javasdk/src/main/java/akka/javasdk/JsonSupport.java

```diff
@@ -123,26 +131,81 @@ public static <T> Any encodeJson(T value, String jsonType) {
     }
   }
 
-  // FIXME do we really want all these to be public API?
+  /**
+   * @deprecated Use encodeToAkkaByteString
+   */
+  @Deprecated
   public static <T> ByteString encodeToBytes(T value) throws JsonProcessingException {
     return UnsafeByteOperations.unsafeWrap(
       objectMapper.writerFor(value.getClass()).writeValueAsBytes(value));
   }
 
-  public static <T> akka.util.ByteString encodeToAkkaByteString(T value) throws JsonProcessingException {
-    return akka.util.ByteString.fromArrayUnsafe(objectMapper.writerFor(value.getClass()).writeValueAsBytes(value));
+  /**
+   * Encode the given value as JSON using Jackson.
+   *
+   * @param value    the object to encode as JSON, must be an instance of a class properly annotated
+   *                 with the needed Jackson annotations.
+   * @throws IllegalArgumentException if the given value cannot be turned into JSON
+   */
+  public static <T> akka.util.ByteString encodeToAkkaByteString(T value) {
+    try {
+      return akka.util.ByteString.fromArrayUnsafe(objectMapper.writerFor(value.getClass()).writeValueAsBytes(value));
+    } catch (JsonProcessingException ex) {
```

Comment:
wrapping the JsonProcessingException is not fully compatible change, but very inconvenient to have that as checked exception, and it's not like that in other places